apps/aecore/src/aec_conductor.erl

```diff
@@ -1338,11 +1359,37 @@ hc_create_block_fun(ConsensusModule, TopHash) ->
         end
     end.
 
-hc_create_block(ConsensusModule, TopHash0, Leader) ->
-    VoteResult = ConsensusModule:vote_result(),
-    TopHash = hc_create_microblock(ConsensusModule, TopHash0, Leader, VoteResult),
-    Res = aec_block_key_candidate:create(TopHash, Leader, Leader),
-    Res.
+%% For as long as chain length is shorter than currentHeight-1, create holes. Send holes async to conductor for writing
+hc_create_hole(TopHash, MissingBlocksCount, Producer) when MissingBlocksCount > 0 ->
+    %% Not creating a microblock, unlike regular hc_create_block
+    GprocKey = {n, l, {production_worker, TopHash, Producer}},
+    case gproc:where(GprocKey) of
+        undefined ->
+            gproc:reg(GprocKey),
+            Result = aec_block_hole_candidate:create(TopHash, Producer, Producer),
+            gproc:unreg(GprocKey),
+            Result;
+        _Pid ->
+            %% Production is already running in another worker, we cannot start a duplicate
+            lager:debug("Busy, try again: Production by ~w already running at ~w", [Producer, _Pid]),
+            {error, busy_try_again}
+    end.
```

Comment:
This looks very hacky, and also like the wrong solution to a problem we shouldn't have? We shouldn't try to create holes/blocks in parallel?

Also, is this actually a problem? I couldn't find `Busy, try again` in the logs after running the tests...

Reply:
The conductor calls the consensus repeatedly with very small intervals, as there is no PoW.
I think at some point from logs we've (me and Richard) observed that it does something weird and one possible explanation was this (sealing more than one block in parallel). I can roll back/remove this one and see if it still works, and that's great if it does.